src/traces/indicator/plot.js

```diff
@@ -0,0 +1,788 @@
+                    titleX = size.l - cn.bulletPadding * size.w; // Outside domain, on the left
+                }
+            } else {
+                // position above numbers
+                titleY = (trace._numbersTop - titlePadding) - titlebBox.bottom;
+            }
+            return strTranslate(titleX, titleY);
+        });
+    });
+};
+
+function drawBulletGauge(gd, plotGroup, cd, gaugeOpts) {
+    var trace = cd[0].trace;
+
+    var bullet = gaugeOpts.bulletGauge;
+    var bulletaxis = gaugeOpts.bulletaxisLayer;
+    var gaugeBg = gaugeOpts.gaugeBg;
+    var gaugeOutline = gaugeOpts.gaugeOutline;
+    var size = gaugeOpts.size;
+    var domain = trace.domain;
+
+    var hasTransition = gaugeOpts.hasTransition;
+    var transitionOpts = gaugeOpts.transitionOpts;
+    var onComplete = gaugeOpts.onComplete;
+
+    // preparing axis
+    var ax, vals, transFn, tickSign, shift;
+    var opts = trace.gauge.axis;
+
+    // Enter bullet, axis
+    bullet.enter().append('g').classed('bullet', true);
+    bullet.attr('transform', 'translate(' + size.l + ', ' + size.t + ')');
+
+    bulletaxis.enter().append('g')
+        .classed('bulletaxis', true)
+        .classed('crisp', true);
+    bulletaxis.selectAll('g.' + 'xbulletaxis' + 'tick,path').remove();
+
+    // Draw bullet
+    var bulletHeight = size.h; // use all vertical domain
+    var innerBulletHeight = trace.gauge.value.thickness * bulletHeight;
+    var bulletLeft = domain.x[0];
+    var bulletRight = domain.x[0] + (domain.x[1] - domain.x[0]) * ((trace._hasNumber || trace._hasDelta) ? (1 - cn.bulletNumberDomainSize) : 1);
+
+    ax = mockAxis(gd, opts, [trace.vmin, trace.vmax]);
+    ax._id = 'xbulletaxis';
+    ax.domain = [bulletLeft, bulletRight];
+    ax.setScale();
+
+    vals = Axes.calcTicks(ax);
+    transFn = Axes.makeTransFn(ax);
+    tickSign = Axes.getTickSigns(ax)[2];
+
+    shift = size.t + size.h;
+    if(ax.visible) {
+        Axes.drawTicks(gd, ax, {
+            vals: ax.ticks === 'inside' ? Axes.clipEnds(ax, vals) : vals,
+            layer: bulletaxis,
+            path: Axes.makeTickPath(ax, shift, tickSign),
+            transFn: transFn
+        });
+
+        Axes.drawLabels(gd, ax, {
+            vals: vals,
+            layer: bulletaxis,
+            transFn: transFn,
+            labelFns: Axes.makeLabelFns(ax, shift)
+        });
+    }
+
+    function drawRect(s) {
+        s
+            .attr('width', function(d) { return Math.max(0, ax.c2p(d.range[1] - d.range[0]));})
+            .attr('x', function(d) { return ax.c2p(d.range[0]);})
+            .attr('y', function(d) { return 0.5 * (1 - d.thickness) * bulletHeight;})
+            .attr('height', function(d) { return d.thickness * bulletHeight; });
+    }
+
+    // Draw bullet background, steps
+    var boxes = [gaugeBg].concat(trace.gauge.steps);
+    var targetBullet = bullet.selectAll('g.targetBullet').data(boxes);
```

Comment:
No :camel:s in class names, please.

https://stackoverflow.com/questions/1547986/can-i-use-camel-case-in-css-class-names

Reply:
> Oh one more thing,
> 
> ![Peek 2019-06-25 16-29](https://user-images.githubusercontent.com/6675409/60131048-73657c80-9766-11e9-8a6e-cefde6d65ccf.gif)
> 
> the "Toggle show closest data on hover" modebar button shouldn't not appear when only `indicator` traces are present on the graph.

What would be the cleanest to achieve that?

Reply:
Thanks so much for noticing this! Camelcases removed in e6d8525ab07c56ca10bdbac6762b06d1d57bf830